src/A2A.AspNetCore/A2AEndpointRouteBuilderExtensions.cs

```diff
@@ -37,10 +37,10 @@ public static IEndpointConventionBuilder MapA2A(this IEndpointRouteBuilder endpo
 
         var routeGroup = endpoints.MapGroup("");
 
-        routeGroup.MapGet(".well-known/agent.json", (HttpRequest request, CancellationToken cancellationToken) =>
+        routeGroup.MapGet(".well-known/agent.json", async (HttpRequest request, CancellationToken cancellationToken) =>
         {
             var agentUrl = $"{request.Scheme}://{request.Host}{path}";
-            var agentCard = taskManager.OnAgentCardQuery(agentUrl, cancellationToken);
+            var agentCard = await taskManager.OnAgentCardQuery(agentUrl, cancellationToken);
```

Comment:
Should this method be named OnAgentCardQueryAsync?

Reply:
Great question! I've created an issue to track it: https://github.com/a2aproject/a2a-dotnet/issues/88 as we have more similar methods